[FSDP][state_dict] Make optim_state_dict_to_load work with use_orig_param=False + NO_SHARD

wz337 opened the pull request:
Summary: As title

Test Plan: CI

Reviewed By: wz337

Differential Revision: D48329724

xunnanxu commented on torch/distributed/fsdp/_optim_utils.py:
```diff
@@ -634,11 +634,17 @@ def _flatten_optim_state(
-                fsdp_state.world_size,
-            )
+            if (
+                fsdp_state.world_size == 1
```
Mind adding some comment around this check?
e.g. what this `== 1` check is for and why is it not `!= 1`?

fegin replied:
Correct, this was a mistake.

fegin committed:
[FSDP][state_dict] Make optim_state_dict_to_load work with use_orig_param=False + NO_SHARD (#107185)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/107185

As title

Test Plan: CI

Reviewed By: wz337

Differential Revision: D48329724

fbshipit-source-id: 5ad491d194039bdc05d14f1feaab2bfa19945af4

fegin commented:
@pytorchbot merge
